.github/workflows/playground-preview.yml

```diff
@@ -1,102 +1,169 @@
         with:
-          name: ${{ vars.PLUGIN_NAME || 'bluehost-wordpress-plugin' }}
-          path: ./artifact
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          run-id: ${{ github.event.workflow_run.id }}
+          fetch-depth: 1
+          persist-credentials: false
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@bf6b4fbd49ca58e4608c9c89fba0b8d90bd2a39f  # v2.35.5
+        with:
+          php-version: '8.1'
+          coverage: none
+          tools: composer, cs2pr
+
+      - name: Use Node from .nvmrc
+        uses: actions/setup-node@a0853c24544627f65ddf259abe73b1d18a591444 # v5.0.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'package-lock.json'
+
+      - name: Setup workflow context
+        id: workflow
+        working-directory: ${{ runner.temp }}
+        env:
+          PLUGIN_NAME: ${{ vars.PLUGIN_NAME }}
+        run: |
+          mkdir dist
+          echo "DIST=${PWD}/dist" >> "$GITHUB_OUTPUT"
+          echo "PACKAGE=$PLUGIN_NAME" >> "$GITHUB_OUTPUT"
+
+      - name: Install PHP Dependencies
+        run: composer install --no-progress --no-dev --optimize-autoloader --prefer-dist
+
+      - name: NPM Install
+        run: npm install
+
+      - name: Build JavaScript
+        run: npm run build
+
+      - name: Prepare files
+        env:
+          DIST: ${{ steps.workflow.outputs.DIST }}
+        run: |
+          rsync -r --include-from=.distinclude --exclude-from=.distignore . "$DIST"
+
+      - name: List dist files (debug)
+        env:
+          DIST: ${{ steps.workflow.outputs.DIST }}
+        run: |
+          set -euo pipefail
+          cd -- "$DIST"
+          find .
 
       - name: Prepare plugin zip for GitHub Pages
         id: prepare
         env:
-          PR_NUMBER: ${{ steps.pr.outputs.number }}
+          DIST: ${{ steps.workflow.outputs.DIST }}
+          PR_NUMBER: ${{ steps.ctx.outputs.number }}
         run: |
-          # Create previews directory structure
+          set -euo pipefail
+
           mkdir -p gh-pages/previews
 
-          # Create zip named by PR number only (overwrites on new pushes)
-          PR_NUM="${PR_NUMBER}"
+          PR_NUM="$PR_NUMBER"
           ZIP_NAME="bluehost-pr-${PR_NUM}.zip"
 
-          # Create zip from artifact directory
-          cd artifact
-          zip -r -9 "../gh-pages/previews/${ZIP_NAME}" .
-          cd ..
+          # Create excludes.txt inside the DIST folder
+          cat > "$DIST/excludes.txt" <<'EOF'
+          .env
+          .env.*
+          .npmrc
+          .yarnrc*
+          auth.json
+          composer.auth.json
+          wp-config.php
+          .git*
+          **/.git/**
+          **/node_modules/**
+          **/.github/**
+          **/.vscode/**
+          **/.idea/**
+          EOF
+
+          # Zip inside DIST using the local excludes.txt file
+          cd "$DIST"
+          zip -r -9 "../${ZIP_NAME}" . -x@excludes.txt
```

Comment:
Is there a reason we're not reusing the .distignore file in the repo? Is the list different? Maybe we can compare and consider using or merging?